### ISSUE TITLE
MS3 fails in the close channel task

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden#5422]
## Problem Definition

Scenario in a nutshell:
node0 : `0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3`
node1 : `0xF95990a11DDF59f9c345a4B23141eC27212d49aB`

```
{
  "task": "<CloseChannelTask: {'from': 0, 'to': 1}>",
  "event": "Task errored",
  "logger": "scenario_player.tasks.base",
  "level": "error",
  "timestamp": "2019-12-03 02:35:51.086747",
  "exception": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 80, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 70, in create_connection\n    sock.connect(sa)\n  File \"/usr/local/lib/python3.7/site-packages/gevent/_socket3.py\", line 335, in connect\n    raise error(err, strerror(err))\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 355, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output\n    self.send(msg)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send\n    self.connect()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 183, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 169, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e)\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 399, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=33845): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0xF95990a11DDF59f9c345a4B23141eC27212d49aB (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 51, in _run\n    method=self._method, url=url, json=self._request_params, timeout=self._timeout\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send\n    raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=33845): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0xF95990a11DDF59f9c345a4B23141eC27212d49aB (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__\n    return self._run(*args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 56, in _run\n    raise RESTAPIError(f\"Error performing REST-API call: {self._name}\") from ex\nscenario_player.exceptions.legacy.RESTAPIError: Error performing REST-API call: close_channel"
}
```

1. Fails while performing REST API call to close the channel
`/api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0xF95990a11DDF59f9c345a4B23141eC27212d49aB`
where 
`0x62083c80353Df771426D209eF578619EE68D5C7A`: token address
`0xF95990a11DDF59f9c345a4B23141eC27212d49aB`: node1

2. Fails at 
`- close_channel: {from: 0, to: 1}`

[ms3_simple_monitoring_2019_12_03.tar.gz](https://github.com/raiden-network/raiden/files/3915418/ms3_simple_monitoring_2019_12_03.tar.gz)

**Comment by rakanalh** [comment]
The exception looks as follows:

```
Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 80, in create_connection
    raise err
  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 70, in create_connection
    sock.connect(sa)
  File \"/usr/local/lib/python3.7/site-packages/gevent/_socket3.py\", line 335, in connect
    raise error(err, strerror(err))
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 603, in urlopen
    chunked=chunked)
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 355, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output
    self.send(msg)
  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send
    self.connect()
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 183, in connect
    conn = self._new_conn()
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 169, in _new_conn
    self, \"Failed to establish a new connection: %s\" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send
    timeout=timeout
  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 641, in urlopen
    _stacktrace=sys.exc_info()[2])
  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 399, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=33845): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0xF95990a11DDF59f9c345a4B23141eC27212d49aB (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 51, in _run
    method=self._method, url=url, json=self._request_params, timeout=self._timeout
  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send
    r = adapter.send(request, **kwargs)
  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=33845): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0xF95990a11DDF59f9c345a4B23141eC27212d49aB (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb4af8184e0>: Failed to establish a new connection: [Errno 111] Connection refused'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__
    return self._run(*args, **kwargs)
  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/execution.py\", line 41, in _run
    task()
  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__
    return self._run(*args, **kwargs)
  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 56, in _run
    raise RESTAPIError(f\"Error performing REST-API call: {self._name}\") from ex
scenario_player.exceptions.legacy.RESTAPIError: Error performing REST-API call: close_channel"}
```

Which happened at:
```
{"result": "scenario error", "event": "Run finished", "logger": "scenario_player.main", "level": "error", "timestamp": "2019-12-03 02:35:51.092787"}
```

But node 0, started at port `33845` at the point in time 
```
{"host": "127.0.0.1", "port": 33845, "node": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3", "event": "REST API starting", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:30:57.249394"}

```
was up and running, and some logs are indicating so:
```
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:48.080731"}
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:48.583739"}
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:49.087505"}
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:49.590461"}
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:50.093477"}
{"response": {"our_address": "0x687Dc97091EE7679be02DFfbd02f48e28f955Bb3"}, "status_code": 200, "event": "Request successful", "logger": "raiden.api.rest", "level": "debug", "timestamp": "2019-12-03 02:35:50.596704"}
```

**Comment by nlsdfnbch** [comment]
unfortunately no. The droplets had no metrics enabled, so there's no way to know for us as of now.

I did update the droplets just now to start pumping memory metrics, so we should be able to see if this occurrs again in the future.

That's the best I can do for now.

**Comment by christianbrb** [comment]
@agatsoh to add logs from Saturday
@nlsdfnbch to analyze the droplet memory metrics

**Comment by nlsdfnbch** [comment]
Metrics show no indication of an OOM event - it may be possible that the docker container, in which each scenario is run, may have further restrictions to max memory use. Will check and get back to you

**Comment by agatsoh** [comment]
On 7thDecember 2019 the MS1 scenarios failed with a same error

```
{
  "task": "<CloseChannelTask: {'from': 0, 'to': 1}>",
  "event": "Task errored",
  "logger": "scenario_player.tasks.base",
  "level": "error",
  "timestamp": "2019-12-07 05:38:50.704065",
  "exception": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 80, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 70, in create_connection\n    sock.connect(sa)\n  File \"/usr/local/lib/python3.7/site-packages/gevent/_socket3.py\", line 335, in connect\n    raise error(err, strerror(err))\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 355, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output\n    self.send(msg)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send\n    self.connect()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 183, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 169, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e)\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 399, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=36669): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 51, in _run\n    method=self._method, url=url, json=self._request_params, timeout=self._timeout\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send\n    raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=36669): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__\n    return self._run(*args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 56, in _run\n    raise RESTAPIError(f\"Error performing REST-API call: {self._name}\") from ex\nscenario_player.exceptions.legacy.RESTAPIError: Error performing REST-API call: close_channel"
}
```

1. Fails at the rest call:
`/api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95`


2. Fails at
`- close_channel: {from: 0, to: 1}`


[ms1_simple_monitoring_2019_12_07.tar.gz](https://github.com/raiden-network/raiden/files/3939215/ms1_simple_monitoring_2019_12_07.tar.gz)

**Comment by agatsoh** [comment]
On 7thDecember 2019 BF1 fails in closechanneltask


```
{
  "task": "<CloseChannelTask: {'from': 3, 'to': 4}>",
  "event": "Task errored",
  "logger": "scenario_player.tasks.base",
  "level": "error",
  "timestamp": "2019-12-07 03:56:58.996153",
  "exception": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 80, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 70, in create_connection\n    sock.connect(sa)\n  File \"/usr/local/lib/python3.7/site-packages/gevent/_socket3.py\", line 335, in connect\n    raise error(err, strerror(err))\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 355, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output\n    self.send(msg)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send\n    self.connect()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 183, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 169, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e)\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fe7c4394128>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 399, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=35385): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x1A797bA53331E92d5319B1fDE6E7D2E34aB11985 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe7c4394128>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 51, in _run\n    method=self._method, url=url, json=self._request_params, timeout=self._timeout\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send\n    raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=35385): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x1A797bA53331E92d5319B1fDE6E7D2E34aB11985 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe7c4394128>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__\n    return self._run(*args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 56, in _run\n    raise RESTAPIError(f\"Error performing REST-API call: {self._name}\") from ex\nscenario_player.exceptions.legacy.RESTAPIError: Error performing REST-API call: close_channel"
}
```

1. Fails while calling the API:

`/api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x1A797bA53331E92d5319B1fDE6E7D2E34aB11985`

2. Fails at 

`- close_channel: {from: 3, to: 4}`


https://drive.google.com/drive/folders/13DV21NX3X_vKUOpRIobhQUA0w8sx1vs6?usp=sharing

**Comment by Dominik1999** [comment]
> On 7thDecember 2019 the MS1 scenarios failed with a same error
> 
> ```
> {
>   "task": "<CloseChannelTask: {'from': 0, 'to': 1}>",
>   "event": "Task errored",
>   "logger": "scenario_player.tasks.base",
>   "level": "error",
>   "timestamp": "2019-12-07 05:38:50.704065",
>   "exception": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 80, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py\", line 70, in create_connection\n    sock.connect(sa)\n  File \"/usr/local/lib/python3.7/site-packages/gevent/_socket3.py\", line 335, in connect\n    raise error(err, strerror(err))\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 355, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1229, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1275, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1224, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 1016, in _send_output\n    self.send(msg)\n  File \"/usr/local/lib/python3.7/http/client.py\", line 956, in send\n    self.connect()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 183, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connection.py\", line 169, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e)\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py\", line 399, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=36669): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 51, in _run\n    method=self._method, url=url, json=self._request_params, timeout=self._timeout\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/requests/adapters.py\", line 516, in send\n    raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=36669): Max retries exceeded with url: /api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f36792df940>: Failed to establish a new connection: [Errno 111] Connection refused'))\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__\n    return self._run(*args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 56, in _run\n    raise RESTAPIError(f\"Error performing REST-API call: {self._name}\") from ex\nscenario_player.exceptions.legacy.RESTAPIError: Error performing REST-API call: close_channel"
> }
> ```
> 
> 1. Fails at the rest call:
>    `/api/v1/channels/0x62083c80353Df771426D209eF578619EE68D5C7A/0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95`
> 2. Fails at
>    `- close_channel: {from: 0, to: 1}`
> 
> [ms1_simple_monitoring_2019_12_07.tar.gz](https://github.com/raiden-network/raiden/files/3939215/ms1_simple_monitoring_2019_12_07.tar.gz)

Ok, that is also due to the transport04 error. Node1 had it and then Node0 got killed

``` 
{
  "current_user": "@0x62c787a0d194c9c7738b53b6e6ee3f1d2a0dbd95:transport04.raiden.network",
  "node": "0x62c787a0d194C9C7738b53b6e6Ee3f1D2a0DbD95",
  "transport_uuid": "9a0c4b85-d559-4a40-858c-6e238b4e270a",
  "config": {
    "available_servers": [
      "https://transport01.raiden.network",
      "https://transport02.raiden.network",
      "https://transport03.raiden.network",
      "https://transport04.raiden.network"
    ],
    "broadcast_rooms": [
      "discovery",
      "path_finding",
      "monitoring"
    ],
    "retries_before_backoff": 5,
    "retry_interval": 5,
    "server": "auto"
  },
  "event": "Matrix stopped",
  "logger": "raiden.network.transport.matrix.transport",
  "level": "debug",
  "timestamp": "2019-12-07 05:38:47.601624"
}
```

**Comment by Dominik1999** [comment]
This did not happen anymore during last week. You had a fix regarding this error, correct @nlsdfnbch ?
If so, it looks like it worked and we might be able to close this issue

**Comment by Dominik2002** [comment]
Closing this for now, should be fixed by #5498
